<launch>
    <arg name="port" default="/dev/ttyACM0"/>
    <arg name="start_seq" default="43605"/> <!-- 0xAA55 in decimal -->
    <arg name="publish_hz" default="200.0"/>

    <arg name="cmd_mode" default="true"/> <!-- false: raw command, true: physical command -->

    <arg name="A_actual_to_pwm_velocity" default="48.0685" />
    <arg name="b_actual_to_pwm_velocity" default="-67.8897" />
    <arg name="A_pwm_to_velocity" default="0.0205" />
    <arg name="b_pwm_to_velocity" default="1.4077" />
    <arg name="A_actual_to_pwm_steer" default="-1578.5040" />
    <arg name="b_actual_to_pwm_steer" default="-10.8340" />
    <arg name="A_pwm_to_steer" default="-0.0006143554755280822" />
    <arg name="b_pwm_to_steer" default="-0.0068634510651983255" />

    <node name="base_board_node" pkg="base_board" type="node" output="screen">
        <param name="port" value="$(arg port)"/>
        <param name="start_seq" value="$(arg start_seq)"/>
        <param name="publish_hz" value="$(arg publish_hz)"/>

        <param name="cmd_mode" value="$(arg cmd_mode)"/>

        <param name="A_actual_to_pwm_velocity" value="$(arg A_actual_to_pwm_velocity)" />
        <param name="b_actual_to_pwm_velocity" value="$(arg b_actual_to_pwm_velocity)" />
        <param name="A_pwm_to_velocity" value="$(arg A_pwm_to_velocity)" />
        <param name="b_pwm_to_velocity" value="$(arg b_pwm_to_velocity)" />
        <param name="A_actual_to_pwm_steer" value="$(arg A_actual_to_pwm_steer)" />
        <param name="b_actual_to_pwm_steer" value="$(arg b_actual_to_pwm_steer)" />
        <param name="A_pwm_to_steer" value="$(arg A_pwm_to_steer)" />
        <param name="b_pwm_to_steer" value="$(arg b_pwm_to_steer)" />
    </node>
</launch>